server/src/http/feature_refresher.rs

```diff
@@ -392,23 +392,15 @@ impl FeatureRefresher {
                                 }
                             }
                             FeatureError::NotFound => {
-                                warn!("Had a bad URL when trying to fetch features. Removing ourselves");
+                                info!("Had a bad URL when trying to fetch features. Removing ourselves from the list of refresh tasks");
```

Comment:
What happens when upstream is back online? I think 404 response code depends on each server/LB implementation, but maybe one reason for 404 is no pods available in a k8s cluster. 

If we consider the backoff implementation we did in the SDKs I think this is correct as it's aligned with that. I'm just wondering if this behavior changes when edge is still not ready (i.e. before receiving at least 1 successful response from upstream)

Reply:
This sounds relevant, but it's a little early for me to understand exactly what you're saying 🥱 Would you be willing to explain this comment a little more for me?

Reply:
So, every time the same key gets used we do the evaluation in the following order
1. Is key even allowed to access - this is stored in the TokenCache
2a. We have equal or wider access to the token's environment already, if so, we should be able to get features from the feature cache
2b We do not have equal or wider access, add the token to the list of tokens our featurerefresher cares about
3. Once we have features, give them to Yggdrasil to build an EngineState
4. Evaluate toggles if /api/frontend, return toggles if /api/client/features
---
What Gaston is worrying about is whether or not we repeat step 2, if upstream comes back online. From what I gather, we'll attempt to register it for refresh once every hour for normal tokens. I'm starting to think we should change the default here to 5 minutes or so.

For startup tokens, we try once every second until they achieve TokenStatus::Validated, then they too get moved into the refresh once per hour.

Reply:
Right, and by step 2, you're referring to 

> 2. Once we have features, give them to Yggdrasil to build an EngineState

Right? Or 2a/2b? Anyway, it sounds like you've got it covered?